properties/base/union.py

```diff
@@ -157,6 +157,10 @@ def assert_valid(self, instance, value=None):
         valid = super(Union, self).assert_valid(instance, value)
         if not valid:
             return False
+        if value is None:
+            value = instance._get(self.name)
+        if value is None:
```

Comment:
Looks like this could be nested into the block above to avoid an additional check on `value`